src/components/AutoComplete.tsx

```diff
@@ -0,0 +1,288 @@
+    onChange: (value: T) => void;
+    className?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const AutoCompleteContextProvider = createContext<AutoCompleteContext<any> | null>(null);
+
+export function useAutoCompleteContext<T extends { id: string }>(): AutoCompleteContext<T> {
+    const ctx = useContext(AutoCompleteContextProvider) as AutoCompleteContext<T> | null;
+
+    if (!ctx) {
+        throw new Error("Don't use before initialization or outse of `AutoComplete` component");
+    }
+
+    return useMemo(() => ctx, [ctx]);
+}
+
+const StyledText = styled(Text).attrs({
+    size: 's',
+    weight: 'regular',
+    color: gray7,
+})`
+    width: 100%;
+    border-bottom: 1px solid ${gray4};
+    margin: ${gapS} 0;
+`;
+
+const StyledLabel = styled.label`
+    display: inline-flex;
+    flex-wrap: nowrap;
+    align-items: baseline;
+
+    input[type='radio'] {
+        padding: 0;
+        margin: 0;
+        margin-right: ${gapXs};
+    }
+`;
+
+const StyledRadioGroup = styled.div`
+    display: flex;
+    align-items: center;
+    gap: ${gapS};
+`;
+
+export function AutoCompleteRadioGroup<T extends { title: string; value: string }>({
+    items,
+    onChange,
+    name,
+    title,
+    value,
+    className,
+}: AutoCompleteRadioGroupProps<T>) {
+    return (
+        <>
+            {nullable(title, (t) => (
+                <StyledText>{t}</StyledText>
+            ))}
+            <StyledRadioGroup className={className}>
+                {items.map((item) => (
+                    <StyledLabel key={item.title}>
+                        <input
+                            type="radio"
+                            name={name}
+                            id={item.title}
+                            value={item.value}
+                            onChange={() => onChange(item)}
+                            defaultChecked={item.value === value}
+                        />
+                        <Text size="s" color={gray7} as="span">
+                            {item.title}
+                        </Text>
+                    </StyledLabel>
+                ))}
+            </StyledRadioGroup>
+        </>
+    );
+}
+
+function getItemCreator<T>(onChange: (item: T) => void, map: React.MutableRefObject<AutoCompleteSelectedMap<T>>) {
+    return function createRenderItem<T1 extends T>(item: T1, index: number): AutoCompleteRenderItemProps<T> {
+        return {
+            item,
+            index,
+            checked: map.current.has(item),
+            onItemClick: () => onChange(item),
+        };
+    };
+}
+
+export function AutoCompleteList({ title, selected }: AutoCompleteListProps) {
+    const {
+        items,
+        value,
+        renderItem,
+        renderItems: Component,
+        renderState,
+        switchType,
+        popItem,
+        pushItem,
+        map,
+    } = useAutoCompleteContext();
+
+    useEffect(() => {
+        if (selected) {
+            switchType();
```

Comment:
It's not obvious what it is. Also I don't understand why we need to do smth in useEffect to render simple list